config/default/nools-extras.js

```diff
@@ -0,0 +1,314 @@
+    const lmpFromPregnancyFollowUp = getLMPDateFromPregnancyFollowUp(v);
+    if (v.reported_date > mostRecentReportDate && lmpFromPregnancyFollowUp !== '' && lmpFromPregnancyFollowUp !== mostRecentLMP) {
+      mostRecentReportDate = v.reported_date;
+      mostRecentLMP = lmpFromPregnancyFollowUp;
+    }
+  });
+  return mostRecentLMP;
+}
+
+
+function isPregnancyTerminatedByAbortion(contact, report) {
+  const followUps = getSubsequentPregnancyFollowUps(contact, report);
+  const latestFollowup = getNewestReport(followUps, antenatalForms);
+  return latestFollowup && getField(latestFollowup, 'pregnancy_summary.visit_option') === 'abortion';
+}
+
+function isPregnancyTerminatedByMiscarriage(contact, report) {
+  const followUps = getSubsequentPregnancyFollowUps(contact, report);
+  const latestFollowup = getNewestReport(followUps, antenatalForms);
+  return latestFollowup && getField(latestFollowup, 'pregnancy_summary.visit_option') === 'miscarriage';
+}
+
+function isActivePregnancy(contact, report) {
+  if (!isPregnancyForm(report)) return false;
+  const lmpDate = getMostRecentLMPDateForPregnancy(contact, report);
+  return lmpDate > today - MAX_DAYS_IN_PREGNANCY * MS_IN_DAY && //Pregnancy registration in the past 9 months
+    !getSubsequentDeliveries(contact, report, 6 * 7).length && //pregnancy not terminated by delivery in last 6 weeks
+    !getSubsequentPregnancies(contact, report).length &&//pregnancy not terminated by another pregnancy report
+    !isPregnancyTerminatedByAbortion(contact, report) &&//pregnancy not terminated by miscarriage or abortion
+    !isPregnancyTerminatedByMiscarriage(contact, report);
+}
+
+function countANCFacilityVisits(contact, pregnancyReport) {
+  //from pregnancy report: How many times has ${patient_short_name} been to the health facility for ANC? [anc_visits_hf/anc_visits_hf_past/visited_hf_count]
+  //from pregnancy visit report:
+  //Did the woman complete the health facility ANC visit scheduled for ${pregnancy_follow_up_date_recent}? [anc_visits_hf/anc_visits_hf_past/last_visit_attended]
+  //Would you like to report any additional unreported health facility ANC visits? [anc_visits_hf/anc_visits_hf_past/report_other_visits]
+
+  //How many? [anc_visits_hf/anc_visits_hf_past/visited_hf_count]
+  let ancHFVisits = 0;
+  const pregnancyFollowUps = getSubsequentPregnancyFollowUps(contact, pregnancyReport);
+  if (getField(pregnancyReport, 'anc_visits_hf.anc_visits_hf_past')) {
+    ancHFVisits += parseInt(getField(pregnancyReport, 'anc_visits_hf.anc_visits_hf_past.visited_hf_count'));
+  }
+  pregnancyFollowUps.forEach(function (report) {
+    const pastANCHFVisits = getField(report, 'anc_visits_hf.anc_visits_hf_past');
+    if (pastANCHFVisits && pastANCHFVisits.last_visit_attended === 'yes') {
+      ancHFVisits += 1;
+    }
+    if (pastANCHFVisits && pastANCHFVisits.report_other_visits === 'yes') {
+      ancHFVisits += parseInt(pastANCHFVisits.visited_hf_count);
```

Comment:
Is it better to check for NaN condition here instead of at the end? Then the function outputs a best-effort tally instead of one bad form breaking this.

Reply:
Moved `NaN` check to above as you suggested, thanks.